### PR TITLE
Allow editing polls

### PR DESCRIPTION
**Pull request by andybalaam**
Polls you created have an edit button:

image

which launches the edit UI:

image

If someone has voted, when you try to edit you see an error dialog:

image

---
Here's what your changelog entry will look like:

## ✨ Features
 * Allow editing polls ([\#7806](https://github.com/matrix-org/matrix-react-sdk/pull/7806)). Contributed by @andybalaam.

Preview: https://pr7806--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/elements/PollCreateDialog.tsx [hunk 1]
```diff
@@ -97,13 +120,32 @@ export default class PollCreateDialog extends ScrollableBaseModal<IProps, IState
...
+            return {
+                "content": {
+                    "m.new_content": pollStart.content,
+                    "m.relates_to": {
+                        "rel_type": "m.replace",
+                        "event_id": this.props.editingMxEvent.getId(),
+                    },
+                },
+                "type": pollStart.type,
+            };
```
Comment (turt2live) [inline src/components/views/elements/PollCreateDialog.tsx:142]: just a note for myself to consider edits properly in extensible events :(

File: src/utils/EventUtils.ts [hunk 2]
```diff
@@ -59,7 +59,10 @@ export function isContentActionable(mxEvent: MatrixEvent): boolean {
 
 export function canEditContent(mxEvent: MatrixEvent): boolean {
     if (mxEvent.status === EventStatus.CANCELLED ||
-        mxEvent.getType() !== EventType.RoomMessage ||
+        (
+            mxEvent.getType() !== EventType.RoomMessage &&
+            !M_POLL_START.matches(mxEvent.getType())
+        ) ||
```
Comment (turt2live) [inline src/utils/EventUtils.ts:65]: there's far too many inversions happening here - can we get a variable please?

**Review by turt2live — COMMENTED** [review]
overall it seems fine - just a minor code style concern

**Review by gaelledel — APPROVED** [review]

**Review by turt2live — APPROVED** [review]
looks like the CI failure is completely unrelated, but unsure who to report it to.

**Pull request merged**